hassio/src/addon-store/hassio-addon-store.ts

```diff
@@ -52,30 +48,25 @@ const sortRepos = (a: HassioAddonRepository, b: HassioAddonRepository) => {
 class HassioAddonStore extends LitElement {
   @property({ attribute: false }) public hass!: HomeAssistant;
 
+  @property({ attribute: false }) public supervisor!: Supervisor;
+
   @property({ type: Boolean }) public narrow!: boolean;
 
   @property({ attribute: false }) public route!: Route;
 
-  @property({ attribute: false }) private _addons?: HassioAddonInfo[];
-
-  @property({ attribute: false }) private _repos?: HassioAddonRepository[];
-
   @internalProperty() private _filter?: string;
 
   public async refreshData() {
-    this._repos = undefined;
-    this._addons = undefined;
-    this._filter = undefined;
     await reloadHassioAddons(this.hass);
     await this._loadData();
   }
 
   protected render(): TemplateResult {
     const repos: TemplateResult[] = [];
 
-    if (this._repos) {
-      for (const repo of this._repos) {
-        const addons = this._addons!.filter(
+    if (this.supervisor.addon.repositories) {
+      for (const repo of this.supervisor.addon.repositories.sort(sortRepos)) {
+        const addons = this.supervisor.addon.addons.filter(
```

Comment:
Make this a memoize

Reply:
or use the guard directive, with `this.supervisor.addon.addons` and `this.supervisor.addon.repositories`